### PR TITLE
userinfo: fix logout button, add sign out confirm page

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
Change the logout button on the user info page so that it sends you to `/.pomerium/sign_out` where you then have to confirm you want to logout.

## Related issues
Fixes https://github.com/pomerium/internal/issues/790

## Checklist

- [x] reference any related issues
- [ ] updated docs
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

**Review by desimone — COMMENTED** [review]
LGTM

File: ui/src/components/SignOutConfirmPage.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import React, { FC } from "react";
+import { SignOutConfirmPageData } from "src/types";
+
+type SignOutConfirmPageProps = {
+  data: SignOutConfirmPageData;
+};
+const SignOutConfirmPage: FC<SignOutConfirmPageProps> = ({ data }) => {
+  function handleClickCancel(evt: React.MouseEvent) {
+    evt.preventDefault();
+    history.back();
```
Comment (nhayfield) [inline ui/src/components/SignOutConfirmPage.tsx:18]: where does history come from? we dont have a router as far as i know.

File: ui/src/components/Header.tsx [hunk 2]
```diff
@@ -6,28 +6,23 @@ import Button from "@mui/material/Button";
...
-          <form action={signOutUrl}>
-            <CsrfInput csrfToken={csrfToken} />
-            <Button variant="text" color="inherit" type="submit">
-              Logout
-            </Button>
-          </form>
-        ) : (
-          <></>
-        )}
+        <Button variant="text" color="inherit" onClick={handleClickLogout}>
```
Comment (nhayfield) [inline ui/src/components/Header.tsx:23]: This page was completely changed in my other PR but I can merge this in and resolve.

**Review by nhayfield — APPROVED** [review]
There will be some merge conflicts but I can resolve and also test the history.back part.

File: ui/src/components/SignOutConfirmPage.tsx (hunk 1 above)
Comment (calebdoxsey) [reply ui/src/components/SignOutConfirmPage.tsx:18]: history is a standard DOM object: https://developer.mozilla.org/en-US/docs/Web/API/History

**Pull request merged**